### PR TITLE
home-work

### PR DESCRIPTION
**Pull request by ddryzhov**
Added CustomGlobalExceptionHandler class + validation for existing DTO classes

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -45,7 +46,7 @@ public void delete(@PathVariable Long id) {
     }
 
     @PutMapping("/{id}")
-    public BookDto updateBook(@PathVariable Long id, @RequestBody BookDto bookDto) {
+    public BookDto updateBook(@PathVariable Long id, @RequestBody @Valid BookDto bookDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/controller/BookController.java]:
```suggestion
    public BookDto updateBook(@PathVariable @Positive Long id, @RequestBody @Valid BookDto bookDto) {
```

Please, add `@Validated` under the controller. Read about `@Positive and @Validated` annotations

File: src/main/java/mate/academy/bookstore/dto/BookDto.java [hunk 2]
```diff
@@ -10,10 +14,27 @@
 @ToString
 public class BookDto {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/dto/BookDto.java]: Let's use Java record for DTO

File: src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -9,10 +13,26 @@
...
+    @NotBlank(message = "Title cannot be blank")
+    @Size(max = 255, message = "Title can be at most 255 characters long")
     private String title;
+
+    @NotBlank(message = "Author cannot be blank")
+    @Size(max = 255, message = "Author can be at most 255 characters long")
     private String author;
+
+    @NotBlank(message = "ISBN cannot be blank")
+    @Size(max = 13, message = "ISBN can be at most 13 characters long")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java]: Check your changelogs also, size should be the same as in Dto

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's consider to improve your implementation

File: src/main/java/mate/academy/bookstore/dto/BookDto.java [hunk 4]
```diff
@@ -1,19 +1,35 @@
...
+        @Size(max = 255, message = "Title can be at most 255 characters long")
+        String title,
+
+        @NotBlank(message = "Author cannot be blank")
+        @Size(max = 255, message = "Author can be at most 255 characters long")
+        String author,
+
+        @NotBlank(message = "ISBN cannot be blank")
+        @Size(max = 255, message = "ISBN can be at most 255 characters long")
+        String isbn,
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/bookstore/dto/BookDto.java]: Check requirments for isbn it`s can`t be 255 characters

File: src/main/java/mate/academy/bookstore/exception/CustomGlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
+    }
+
+    private String getErrorMessage(ObjectError e) {
+        if (e instanceof FieldError) {
+            String field = ((FieldError) e).getField();
+            String message = e.getDefaultMessage();
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/bookstore/exception/CustomGlobalExceptionHandler.java]:
```suggestion
        if (e instanceof FieldError error) {
            String field = error.getField();
            String message = e.getDefaultMessage();
```

Comment (MentorMateAcademy) [reply src/main/java/mate/academy/bookstore/exception/CustomGlobalExceptionHandler.java]: try this way

File: src/main/java/mate/academy/bookstore/exception/CustomGlobalExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors()
+                .stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/bookstore/exception/CustomGlobalExceptionHandler.java]: Save strings to constants

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

**Review by koval-kn — APPROVED** [review]

**Pull request merged**